### DISCUSSION TITLE
Wizards with branching logic that also maintain context?

### DISCUSSION THREAD

**Discussion by humblecoder** [telegraf/telegraf#1686] [Q&A]
Trying to wrap my head about the best way to implement a branching wizard that maintains context.  Something like branching "choose your own adventure" style wizards, perhaps in "non-monolithic" form (stages/scenes in separate files).  

Thinking something more like a traversable state machine than anything else.  Any examples of this?  🤔

**Comment by MKRhere** [comment]
You can `ctx.wizard.enter(choice)` from any wizard, and branch using that. Does that work?

**Reply by humblecoder** [reply]
That does work, but I was looking for something less "numeric".  I see that `.action('CUSTOM_ACTION_ID')` is actually what I need.

Thanks for the input!

**Reply by humblecoder** [reply]
Out of curiosity, again, assuming I **do not** want a "monolithic" file with every wizard/scene in the same file as the actual bot, what are the best ways to :

- ... pass information/state into the wizard?
- ... successfully **wait, then retrieve** the final state from the wizard in order to act on the information?
